File: SlidingWindow/SubarraySumLessThanEqualK.js
/**
 * @param {number[]} nums
 * @param {number} k
 */
function findLength(nums, k) {
    let left = 0;
    let curr = 0;
    let answer = 0;

    for (let right = 0; right < nums.length; right++) {
        curr += nums[right];

        while (curr > k) {
            curr -= nums[left];
            left++;
        }

        answer = Math.max(answer, curr);
    }

    return answer;
}

findLength([3, 1, 2, 7, 4, 2, 1, 1, 5], 8);

File: SlidingWindow/SubarraySumLessThanEqualK.py
def find_length(nums: list[int], k: int) -> int:
    left, curr, answer = 0, 0, 0

    for right in range(len(nums)):
        curr += nums[right]

        while curr > k:
            curr -= nums[left]
            left += 1
        
        answer = max(answer, right - left + 1)

    return answer


print(find_length([3, 1, 2, 7, 4, 2, 1, 1, 5], 8))